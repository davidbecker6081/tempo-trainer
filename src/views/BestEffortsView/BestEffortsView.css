.best-efforts-view {
  width: 40%;
}
