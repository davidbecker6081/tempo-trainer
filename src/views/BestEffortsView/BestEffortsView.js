import React from 'react';
import PropTypes from 'prop-types';
import BestEffort from '../../components/BestEffort/BestEffort';
import './BestEffortsView.css';

const BestEffortsView = ({ channelSet, dataHelper }) => {
  const efforts = {
    1: dataHelper.calculateBestEffort(channelSet, 1),
    5: dataHelper.calculateBestEffort(channelSet, 5),
    10: dataHelper.calculateBestEffort(channelSet, 10),
    15: dataHelper.calculateBestEffort(channelSet, 15),
    20: dataHelper.calculateBestEffort(channelSet, 20),
  };

  return (
    <section>
      <BestEffort
        channelSet={channelSet}
        timeForCalc={1}
        timeRange={efforts['1'].range}
        effort={efforts['1'].average}
      />
      <BestEffort
        channelSet={channelSet}
        timeForCalc={5}
        timeRange={efforts['5'].range}
        effort={efforts['5'].average}
      />
      <BestEffort
        channelSet={channelSet}
        timeForCalc={10}
        timeRange={efforts['10'].range}
        effort={efforts['10'].average}
      />
      <BestEffort
        channelSet={channelSet}
        timeForCalc={15}
        timeRange={efforts['15'].range}
        effort={efforts['15'].average}
      />
      <BestEffort
        channelSet={channelSet}
        timeForCalc={20}
        timeRange={efforts['20'].range}
        effort={efforts['20'].average}
      />
    </section>
  );
};

export default BestEffortsView;

BestEffortsView.propTypes = {
  dataHelper: PropTypes.shape({
    data: PropTypes.shape({}),
    channels: PropTypes.shape([]),
    GPSCoords: PropTypes.shape([]),
    min: PropTypes.number,
    max: PropTypes.number,
  }).isRequired,
};
