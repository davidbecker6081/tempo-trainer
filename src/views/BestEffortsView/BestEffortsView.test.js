import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import BestEffortsView from './BestEffortsView';
import DataHelper from '../../components/DataCleaner/DataCleaner';
import workoutData from '../../__mock__/workout-data.json';

describe('BestEffortsView', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      channelSet: '',
      dataHelper: {},
    };
  });

  describe('channelSet is defined', () => {
    beforeEach(() => { mockProps.channelSet = 'power'; });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('dataHelper is defined', () => {
    beforeEach(() => { mockProps.dataHelper = new DataHelper(workoutData); });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
