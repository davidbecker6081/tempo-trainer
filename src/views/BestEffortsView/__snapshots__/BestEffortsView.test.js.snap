// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BestEffortsView channelSet is defined should render correclty 1`] = `undefined`;

exports[`BestEffortsView dataHelper is defined should render correclty 1`] = `undefined`;
